src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,147 @@
+            if (input.length() > DONE_SUFFIX_LENGTH) {
+                try {
+                    int taskIndex = Integer.parseInt(
+                            input.substring(DONE_SUFFIX_LENGTH)
+                    );
+                    Task item = this.taskList.get(taskIndex - 1);
+                    item.setDone();
+                    return ("Nice! I've marked this task as done:\n  " + item);
+                } catch (NumberFormatException e) {
+                    throw new DukeException("The description of a done"
+                            + " must be an integer.");
+                }
+            } else {
+                throw new DukeException("The description of a done"
+                        + " cannot be empty.");
+            }
+        } else if (input.startsWith("find ")) {
+            if (input.length() > FIND_SUFFIX_LENGTH) {
+                String regex = input.substring(FIND_SUFFIX_LENGTH);
+                return this.taskList.getSearchList(regex);
+            } else {
+                throw new DukeException("The description of a find"
+                        + " cannot be empty");
+            }
+        } else if (input.startsWith("delete ")) {
+            if (input.length() > DELETE_SUFFIX_LENGTH) {
+                try {
+                    int taskIndex = Integer.parseInt(
+                            input.substring(DELETE_SUFFIX_LENGTH)
+                    );
+                    if (this.taskList.size() >= taskIndex && taskIndex > 0) {
+                        String output = ("Noted. I've removed this task:\n  "
+                                + this.taskList.get(taskIndex - 1)
+                                + "\nNow you have "
+                                + (this.taskList.size() - 1)
+                                + " tasks in the list.");
+                        this.taskList.remove(taskIndex - 1);
+                        return output;
+                    } else {
+                        throw new DukeException("The integer entered for "
+                                + "deletion is not valid.");
+                    }
+                } catch (NumberFormatException e) {
+                    throw new DukeException("The description of a delete "
+                            + "must be an integer.");
+                }
+            } else {
+                throw new DukeException("The description of a delete "
+                        + "cannot be empty.");
+            }
+        } else if (input.startsWith("todo")) {
+            return addToList(input, TaskType.Todo);
+        } else if (input.startsWith("deadline ")) {
+            return addToList(input, TaskType.Deadline);
+        } else if (input.startsWith("event ")) {
+            return addToList(input, TaskType.Event);
+        } else if (input.equals("list")) {
+            return this.taskList.getList();
+        } else {
+            throw new DukeException("I'm sorry, but I don't know what"
```

Comment:
Alternatively, you can define another class like that extends duke exception, like InvalidCommandException. So that you can just call it, so you would not need to type the message every time.

Reply:
Good idea, thanks for the suggestion! That would make better OOP sense as well. (: